java/kafka/consumer/src/main/java/KafkaConsumerConfig.java

```diff
@@ -56,7 +76,15 @@ public KafkaConsumerConfig(String bootstrapServers, String topic, String groupId
         this.additionalConfig = additionalConfig;
     }
 
-    public static KafkaConsumerConfig fromEnv() {
+    public static KafkaConsumerConfig fromEnv() { // Populate JAEGER_SERVICE_NAME and OTEL_SERVICE_NAME with User defined JAEGER_SERVICE_NAME
+        // because OpeTracing uses JAEGER_SERVICE_NAME and OpenTelemetry uses OTEL_SERVICE_NAME.
+        String serviceName = System.getenv("OTEL_SERVICE_NAME")  == null ? DEFAULT_SERVICE_NAME : System.getenv("OTEL_SERVICE_NAME");
+        String tracesExporterNames = System.getenv("OTEL_TRACES_EXPORTER")  == null ? DEFAULT_EXPORTER_NAMES : System.getenv("OTEL_TRACES_EXPORTER");
+        String metricsExporterNames = System.getenv("OTEL_METRICS_EXPORTER")  == null ? DEFAULT_METRICS_NAMES : System.getenv("OTEL_METRICS_EXPORTER");
```

Comment:
We still need to figure out if these variables need to be added  to the Kafka client properties files or the process's ENV VARs. If the former we need to set these in the `createProperties` method, if the latter we need to set the ENV VARs of the process programmatically

Reply:
All these variables are not needed at all in the clients. They are just for the OpenTelemetry framework.
It's needed that the user set them when starting the client example but the client code itself doesn't need them.
Also the "metrics" part is something different from tracing. You just need the traces exporter not the metrics one.
At same time the OpenTelemetry SDK auto-configuration extension works if you set the metrics exporter and in our case it has to be `none` but you can do in the code setting the corresponding system property directly `otel.metrics.exporter`.
See here for reference: https://github.com/strimzi/strimzi-kafka-bridge/blob/ad424852e0c741d5404a164e6eba1f249c68baa0/src/main/java/io/strimzi/kafka/bridge/tracing/OpenTelemetryHandle.java#L67

Reply:
Return to this